client/notifications/notification-reducer.ts

```diff
@@ -93,17 +98,23 @@ export default keyedReducer(new NotificationState(), {
       .deleteIn(['notificationIds', notificationId])
   },
 
-  ['@notifications/clearBegin'](state) {
+  ['@notifications/clearBegin'](state, { payload: { timestamp } }) {
+    clearedIdToNotification = state.idToNotification.filter(n => n.createdAt <= timestamp)
```

Comment:
Since local notifications don't have a server-relative timestamp (but also don't have the weirdness of time delay between sending/receiving), you should always clear all local notifications here and not try to compare times

Reply:
You also need to be careful here, what if someone clicks the button 5 times before the server responds, and then the server responds with an error? A better solution here would be to track the request through a field on the action (e.g. give each request an ID, store these optimistically-removed items grouped under that ID, clear them out of this structure once you see a `@notifications/clear` with that id)